src/bin/laptop_scrapper.rs

```diff
@@ -0,0 +1,474 @@
+                .await?;
+            }
+        }
+
+        c.close_window().await?;
+        c.close().await?;
+        Ok(())
+    }
+    .map(|result| {
+        if result.is_err() {
+            println!("{result:#?}");
+        }
+        result
+    })
+    .boxed()
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct WebDriverSettings {
+    pub host: String,
+    pub port: u16,
+}
+
+impl Default for WebDriverSettings {
+    fn default() -> Self {
+        Self{host: String::from("127.0.0.1"), port: 9515}
+    }
+}
+
+impl WebDriverSettings {
+    fn connection_url(self) -> String {
+        format!("http://{}:{}", self.host, self.port)
+    }
+}
+
+pub fn get_configuration() -> Result<WebDriverSettings, config::ConfigError> {
+    config::Config::builder()
+        .add_source(config::Config::try_from(&WebDriverSettings::default()).unwrap())
+        .add_source(config::File::with_name("webdriver.yaml"))
+        .add_source(
+            config::Environment::with_prefix("LAPTOP_SCRAPPER")
+                .try_parsing(true)
+                .separator("_"),
+        )
+        .build()?
+        .try_deserialize()
+}
+
+
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    let webdriver_url = get_configuration()?.connection_url();
+    let pool = Arc::new(connect().await);
+    let semaphore = Arc::new(Semaphore::new(10));
+
+    let mut set = tokio::task::JoinSet::new();
+
+    let mut cpus = Arc::new(get_cpus(pool.clone()).await?);
+    if cpus.is_empty() {
+        set.spawn(parse(
+            webdriver_url.clone(),
+            String::from("https://www.cpubenchmark.net/cpu_list.php"),
+            ParserType::CpuBenchmark,
+            pool.clone(),
+            semaphore.clone(),
+        ));
+    }
+
+    let mut gpus = Arc::new(get_gpus(pool.clone()).await?);
+    if gpus.is_empty() {
+        set.spawn(parse(
+            webdriver_url.clone(),
+            String::from("https://www.videocardbenchmark.net/gpu_list.php"),
+            ParserType::GpuBenchmark,
+            pool.clone(),
+            semaphore.clone(),
+        ));
+    }
+
+    while let Some(result) = set.join_next().await {
+        if result.is_err() {
+            println!("{result:#?}");
+        }
+    }
+
+    // All data is saved to database
+    if cpus.is_empty() {
+        cpus = get_cpus(pool.clone()).await?.into();
+    }
+
+    if gpus.is_empty() {
+        gpus = get_gpus(pool.clone()).await?.into();
+    }
+
+    let laptops = get_laptops(pool.clone()).await?;
+    if laptops.is_empty() {
+        set.spawn(parse(
+            webdriver_url.clone(),
+            String::from("https://rozetka.com.ua/ua/notebooks/c80004/"),
+            ParserType::RozetkaLaptopList(true, cpus, gpus),
+            pool.clone(),
+            semaphore.clone(),
+        ));
+        if let Some(result) = set.join_next().await {
+            if result.is_err() {
+                println!("{result:#?}");
+            }
+        };
```

Comment:
I am not sure, but maybe it can be written with the [transpose](https://doc.rust-lang.org/std/option/enum.Option.html#method.transpose) method into something like this

```rust
if set.join_next().await.transpose().is_err() { ...  }
```

Or

```rust
if let Err(err) = set.join_next().await.transpose() {
       println!("{result:#?}");
};
```